tests/cpptests/test_cpp_vector_filter.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+#include "src/redisearch.h"
+#include "src/spec.h"
+#include "query_test_utils.h"
+
+#include "gtest/gtest.h"
+
+#include <algorithm>
+
+class VectorFilterTest : public ::testing::Test {
+protected:
+  void SetUp() override {
+  }
+};
+
+bool isValidAsVectorFilter(const char *qt, RedisSearchCtx &ctx) {
+  QASTCXX ast;
+  ast.setContext(&ctx);
+  return ast.isValidAsVectorFilter(qt);
+}
+
+#define assertValidVectorFilter(qt, ctx) ASSERT_TRUE(isValidAsVectorFilter(qt, ctx))
+#define assertInvalidVectorFilter(qt, ctx) ASSERT_FALSE(isValidAsVectorFilter(qt, ctx))
+
+TEST_F(VectorFilterTest, testVectorFilter) {
```

Comment:
small suggestion, why don't we put the ctx construction in the setup, and then you can separate in different tests where you test the different type of validations?